x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/step_rule_actions/index.tsx

```diff
@@ -101,7 +101,7 @@ const StepRuleActionsComponent: FC<StepRuleActionsProps> = ({
     [actionMessageParams, summaryActionMessageParams]
   );
   const displayResponseActionsOptions = useMemo(() => {
-    if (isQueryRule(ruleType)) {
+    if (isQueryRule(ruleType) || isEsqlRule(ruleType) || isEqlRule(ruleType)) {
```

Comment:
Thanks Maxim, good point! I am afraid we cannot call response actions against all the rules types because in order to do so we need to know what agent the alert is referring to. Do you know if any other alerts (from other rules than Query, E(S)QL contain `agent.id` of any kind?

Reply:
For now I'll adjust to use `shouldShowResponseActions` , but I'll drop the checks if you say we can support more types.